mlflow/models/model.py

```diff
@@ -22,18 +22,18 @@ class Model(object):
     """
 
     def __init__(self, artifact_path=None, run_id=None, utc_time_created=None, flavors=None,
-                 signature: ModelSignature = None, input_example: ModelInputExample = None,
+                 signature: ModelSignature=None, input_example: Dict[str, Any]=None,
```

Comment:
Why change this type away from `ModelInputExample`?

Reply:
The input example here is example metadata rather than the actual example. I admit it is confusing. I can rename it or create a type for it to make it more clear.

Reply:
Thanks for clarifying, maybe we can also add a docstring for the constructor to explain what the passed-in values are? I know there isn't an existing one, but would be nice given that this class [is publicly documented](https://mlflow.org/docs/latest/python_api/mlflow.models.html#mlflow.models.Model)

Reply:
Also happy to do that in a follow-up^, as this PR otherwise LGTM

Reply:
Makes sense!